core/src/nodes/listeners.rs

```diff
@@ -90,15 +93,25 @@ where
 {
     /// The object that actually listens.
     listener: TTrans::Listener,
-    /// Address it is listening on.
-    address: Multiaddr,
+    /// Addresses it is listening on.
+    addresses: HashSet<Multiaddr>
```

Comment:
What about `SmallVec<[Multiaddr; 2]>`, as this is the number of addresses that I'd expect (127.0.0.1 and the interface IP)?

Reply:
Can do. However the number of addresses is clearly dependent on the number of interfaces (and >2 even on my machine). Still small enough I guess so that going from O(1) to O(n) for inserts/lookups does not hurt.